src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -61,8 +62,12 @@ internal static Exception CreateOpenSslCryptographicException()
                 return new CryptographicException();
             }
 
+            // Even though ErrGetError returns ulong (C++ unsigned long), we 
+            // really only expect error codes in the UInt32 range
+            Debug.Assert(error <= uint.MaxValue, "ErrGetError should only return error codes in the UInt32 range.");
+
             // Inline version of the ERR_GET_REASON macro.
-            uint reason = error & ReasonMask;
+            ulong reason = error & ReasonMask;
```

Comment:
> Where is ERR_R_MALLOC_FAILURE below defined?

It is defined in C# at the beginning of the method.

> there should be a static assert in shim that we have the value right.

How about instead ErrGetError returns a "out bool isAllocFailure"?  Then we can get rid of all the special knowledge of OpenSSL values in C# code:  ReasonMask, ERR_R_MALLOC_FAILURE, and the simulation of ERR_GET_REASON macro.  If/when we need to add more, we can change the bool into an enum that we define/control in both managed and the shim code.  The shim code would convert OpenSSL reasons into our own reasons before passing it back to managed code. But since we only need 1 now, we can just use a bool and change it later, if needed. Thoughts?